device/core/src/iotedge_authentication_provider.ts

```diff
@@ -95,14 +97,35 @@ export class IotEdgeAuthenticationProvider extends SharedAccessKeyAuthentication
     // Codes_SRS_NODE_IOTEDGED_AUTHENTICATION_PROVIDER_13_005: [ The constructor shall throw a TypeError if the _authConfig.workloadUri field is not a valid URI. ]
     // Codes_SRS_NODE_IOTEDGED_AUTHENTICATION_PROVIDER_13_006: [ The constructor shall build a unix domain socket path host if the workload URI protocol is unix. ]
     // Codes_SRS_NODE_IOTEDGED_AUTHENTICATION_PROVIDER_13_007: [ The constructor shall build a string host if the workload URI protocol is not unix. ]
-    let workloadUri = url.parse(this._authConfig.workloadUri);
+    this._workloadUri = url.parse(this._authConfig.workloadUri);
     const config: RestApiClient.TransportConfig = {
-      host: workloadUri.protocol === 'unix:' ? { socketPath: workloadUri.pathname } : workloadUri.host
+      host: this._workloadUri.protocol === 'unix:' ? { socketPath: this._workloadUri.pathname } : this._workloadUri.hostname
     };
 
     this._restApiClient = new RestApiClient(config, `${packageJson.name}/${packageJson.version}`);
```

Comment:
>`${packageJson.name}/${packageJson.version}` [](start = 52, length = 44)

nitpick for later: there's actually an util.getUserAgent() function that can be used to comply with useragent v2 requirements. this being said I believe it's async so... maybe not usable in this case.

Reply:
I'll add a todo comment line. One way to handle this might be to make `this._restApiClient` lazily initialized when needed.